### ISSUE TITLE
Crash in ModuleInfo destructor

### ISSUE THREAD

**Issue by mwpenny** [laverdet/isolated-vm#139]
First, thank you for your work on this great module!

After running the following:

```js
const ivm = require('isolated-vm');

const isolate = new ivm.Isolate();
const mod = isolate.compileModuleSync('');

process.stdin.resume();  // Hack to keep the process alive to demonstrate the crash
```

and waiting for ~10-15 seconds, the process crashes with the following stack trace:

```
FATAL ERROR: v8::HandleScope::CreateHandle() Cannot create a handle without a HandleScope
 1: 00007FF6F8D1F67F napi_wrap+121039
 2: 00007FF6F8CC5736 v8::base::CPU::has_sse+34342
 3: 00007FF6F8CC63F6 v8::base::CPU::has_sse+37606
 4: 00007FF6F94B96DC v8::FunctionTemplate::RemovePrototype+284
 5: 00007FF6F939E965 v8::internal::HandleScope::Extend+53
 6: 00007FF6F949E655 v8::HandleScope::CreateHandle+37
 7: 00007FFB617139CA ivm::ModuleInfo::~ModuleInfo+106 [<MY_PROJECT>\node_modules\isolated-vm\src\module_handle.cc]:L33
 8: 00007FFB61715475 ivm::ModuleHandle::`scalar deleting destructor'+69
 9: 00007FF6F93A00ED v8::internal::GlobalHandles::InvokeFirstPassWeakCallbacks+157
10: 00007FF6F937682D v8::internal::Heap::PageFlagsAreConsistent+2397
11: 00007FF6F936C7A3 v8::internal::Heap::CollectGarbage+1235
12: 00007FF6F936F31A v8::internal::Heap::EphemeronKeyWriteBarrierFromCode+1770
13: 00007FF6F9365893 v8::internal::IncrementalMarking::WasActivated+467
14: 00007FF6F8C71D8C v8::internal::wasm::SignatureMap::Freeze+121660
15: 00007FF6F8C70D10 v8::internal::wasm::SignatureMap::Freeze+117440
16: 00007FF6F8D67BDB uv_async_send+331
17: 00007FF6F8D6737C uv_loop_init+1212
18: 00007FF6F8D67544 uv_run+244
19: 00007FF6F8C8E8F2 v8::internal::UnoptimizedCompilationInfo::feedback_vector_spec+29362
20: 00007FF6F8CEA213 node::Start+275
21: 00007FF6F8BAC91C AES_cbc_encrypt+15180
22: 00007FF6F9980FD8 v8::internal::SetupIsolateDelegate::SetupHeap+1288216
23: 00007FFB9B0F84D4 BaseThreadInitThunk+20
24: 00007FFB9B3CE851 RtlUserThreadStart+33
```

I was able to reproduce this on the latest version (2.0.2).

---

This can be encountered in practice if the module exists in a closure outside the isolate for an extended period. In my application, the function in the isolate calls a callback outside the isolate after a promise is resolved - the crash occurs if the promise never resolves, or just takes a long time.

**Comment by mwpenny** [comment]
I am using node v12.10.0

**Comment by appellation** [comment]
I'm also running into this issue on Node v13.2.0. It seems to occur when the isolate is GC'd: I was able to resolve it by calling `Isolate#dispose` before the reference is lost.

**Comment by laverdet** [comment]
Thanks for the easy reproduction case. This is resolved and the fix is on npm now as 2.0.3.
